src/Compilers/CSharp/Test/Syntax/Syntax/FieldAndValueKeywordTests.cs

```diff
@@ -0,0 +1,1241 @@
+                        }
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular12);
+            comp.VerifyEmitDiagnostics(
+                // (8,34): info CS9248: 'field' is a contextual keyword, with a specific meaning, starting in language version preview. Use '@field' to avoid a breaking change when compiling with language version preview or later.
+                //             object F1() { object field = 1; return null; };
+                Diagnostic(ErrorCode.INF_IdentifierConflictWithContextualKeyword, "field = 1").WithArguments("field", "preview").WithLocation(8, 34),
+                // (14,32): info CS9248: 'value' is a contextual keyword, with a specific meaning, starting in language version preview. Use '@value' to avoid a breaking change when compiling with language version preview or later.
+                //             void G1() { object value = 1; }
+                Diagnostic(ErrorCode.INF_IdentifierConflictWithContextualKeyword, "value = 1").WithArguments("value", "preview").WithLocation(14, 32));
+        }
+
+        [Fact]
+        public void LocalFunction_Parameter()
+        {
+            string source = """
+                #pragma warning disable 649, 8321
+                class C
+                {
+                    object P
+                    {
+                        get
+                        {
+                            object F1(object field) => null;
+                            object F2(object @field) => null;
+                            return null;
+                        }
+                        set
+                        {
+                            void G1(object value) { }
+                            void G2(object @value) { }
+                        }
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular12);
+            comp.VerifyEmitDiagnostics(
+                // (8,23): info CS9248: 'field' is a contextual keyword, with a specific meaning, starting in language version preview. Use '@field' to avoid a breaking change when compiling with language version preview or later.
+                //             object F1(object field) => null;
+                Diagnostic(ErrorCode.INF_IdentifierConflictWithContextualKeyword, "object field").WithArguments("field", "preview").WithLocation(8, 23),
+                // (14,21): info CS9248: 'value' is a contextual keyword, with a specific meaning, starting in language version preview. Use '@value' to avoid a breaking change when compiling with language version preview or later.
+                //             void G1(object value) { }
+                Diagnostic(ErrorCode.INF_IdentifierConflictWithContextualKeyword, "object value").WithArguments("value", "preview").WithLocation(14, 21));
+        }
```

Comment:
nit: Consider adding a test for attribute scenarios, even though they are being kept out-of-scope for this PR:
- [MyAttribute(nameof(value))]
- attribute target scenario: `[field: MyAttribute]` and `[@field: MyAttribute]`, on accessors